src/llvm_ir_generator.cpp

```diff
@@ -261,11 +267,26 @@ void LLVMIRGenerator::Visit(const RecordVarDeclNode& record_var_decl) {
   for (auto i = std::size_t{0}, e = record_var_decl.inits.size(),
             slot_count = record_type.SlotCount();
        i < slot_count && i < e; ++i) {
+    // TODO: consider the case for union when there are multiple designators. It
+    // seems that both Clang and gcc have initialized the following union with
+    // the value 6.
+    // union shape s = {.circle = 5, .triangle = 6};
     auto& init = record_var_decl.inits.at(i);
+    record_des_vals.clear();
     init->Accept(*this);
     auto init_val = val_recorder.ValOfPrevExpr();
 
-    auto res_addr = builder_.CreateStructGEP(type, base_addr, i);
+    llvm::Value* res_addr = nullptr;
+    if (record_des_vals.empty()) {
+      res_addr = builder_.CreateStructGEP(type, base_addr, i);
+    } else {
+      // TODO: consider array index designators.
+      // NOTE: Only consider identifer designators for now.
+      assert(record_des_vals.size() == 1);
+      auto mem_index =
+          record_type.MemberIndex(std::get<std::string>(record_des_vals.at(0)));
+      res_addr = builder_.CreateStructGEP(type, base_addr, mem_index);
```

Comment:
The if-condition is to determine the index, thus consider moving `CreateStructGEP` outside of the loop.

```cpp
auto init_index = i;
if (!record_des_vals.empty()) {
  ...
}
res_addr = builder_.CreateStructGEP(type, base_addr, init_index);
...
```